app/scripts/directives/addSecretToApplication.js

```diff
@@ -0,0 +1,180 @@
+        group: 'apps',
+        resource: 'statefulsets'
+      }, context).then(function(statefulSetData) {
+        statefulSets = _.toArray(statefulSetData.by('metadata.name'));
+        sortApplications();
+      });
+    };
+
+    ctrl.$onInit = function() {
+      ctrl.addType = 'env';
+      ctrl.disableInputs = false;
+      getApplications();
+    };
+
+    ctrl.$postLink = function() {
+      $scope.$watch(function() {
+        return ctrl.application;
+      }, function() {
+        // Look at the existing mount paths so that we can warn if the new value is not unique.
+        var podTemplate = _.get(ctrl.application, 'spec.template');
+        ctrl.existingMountPaths = StorageService.getMountPaths(podTemplate);
+      });
+    };
+
+    ctrl.addToApplication = function() {
+      var applicationToUpdate = angular.copy(ctrl.application);
+
+      var podTemplate = _.get(applicationToUpdate, 'spec.template');
+
+      ctrl.disableInputs = true;
+
+      if (ctrl.addType === 'env') {
+        var newEnvFrom = {
+          secretRef: {
+            name: ctrl.secret.metadata.name
+          }
+        };
+
+        // For each container, add the new volume mount.
+        _.each(podTemplate.spec.containers, function(container) {
+          container.envFrom = container.envFrom || [];
+          container.envFrom.push(newEnvFrom);
+        });
+      } else {
+        var generateName = $filter('generateName');
+        var name = generateName(ctrl.secret.metadata.name + '-');
+        var newVolumeMount = {
+          name: name,
+          mountPath: ctrl.mountVolume,
+          readOnly: true
+        };
+
+        // For each selected container, add the new volume mount.
+        _.each(podTemplate.spec.containers, function(container) {
+          container.volumeMounts = container.volumeMounts || [];
+          container.volumeMounts.push(newVolumeMount);
+        });
+
+        var newVolume = {
+          name: name,
+          secret: {
+            secretName: ctrl.secret.metadata.name
+          }
+        };
+
+        podTemplate.spec.volumes = podTemplate.spec.volumes || [];
+        podTemplate.spec.volumes.push(newVolume);
+      }
+
+      var humanizeKind = $filter('humanizeKind');
+      var sourceKind = humanizeKind(ctrl.secret.kind);
+      var targetKind = humanizeKind(applicationToUpdate.kind);
+      var context = {
+        namespace: ctrl.project.metadata.name
+      };
+
+      DataService.update(APIService.kindToResource(applicationToUpdate.kind), applicationToUpdate.metadata.name, applicationToUpdate, context).then(
+        function() {
+          NotificationsService.addNotification({
+            type: "success",
+            message: "Successfully added " + sourceKind + " " + ctrl.secret.metadata.name + " to " + targetKind + " " + applicationToUpdate.metadata.name + ".",
+            links: [{
+              href: Navigate.resourceURL(applicationToUpdate),
+              label: "View " + targetKind
```

Comment:
Nit: better to call `humanizeKind(applicationToUpdate.kind, true)` to get title case. This is consistent with our other notification links.

Reply:
Fixed